operational_analysis/methods/plant_analysis.py

```diff
@@ -713,51 +819,65 @@ def sample_long_term_reanalysis(self, n, r):
+        ws_aggregate.reset_index(drop=True, inplace=True)
+        # Wind speed IAV calculation
+        ws_aggregate_sd = ws_data.groupby(ws_data.index.month)[r].std() # Get long-term annualized stdev of daily wind speeds            
+        if self.time_resolution == 'D':
+            ws_aggregate_sd = np.repeat(ws_aggregate_sd, np.ceil(self.num_days_lt).astype(int))
+        ws_aggregate_sd.reset_index(drop=True,inplace=True)
+        iav_df = ws_aggregate_sd/ws_aggregate
+        iav_df = iav_df.to_frame()
+        iav_df['sample'] = iav_df.apply(lambda row: np.random.normal(1,row[r],1), axis=1)
+        mc_ws_aggregate = ws_aggregate * iav_df['sample']  
+        long_term_reg_inputs = mc_ws_aggregate.astype(float).reset_index(drop=True)
+        
+        # If temperature is an input, sample long-term temperature values
+        if self.reg_temperature == True:
+            temp_df = self._reanalysis_aggregate[r + '_temp'].to_frame().dropna() # Drop NA values from monthly/daily reanalysis data series
+            temp_data = temp_df.tail(n * self._calendar_samples) # Get last 'x' years of data from reanalysis product
+            temp_aggregate = self.groupby_time_res(temp_data)[r + '_temp'] # Get long-term annualized monthly/daily temperatures
+            long_term_reg_inputs = pd.concat([mc_ws_aggregate.astype(float).reset_index(drop=True), temp_aggregate.astype(float).reset_index(drop=True)], axis=1)
+        
+        # If wind direction is an input, sample long-term wind direction values
+        if self.reg_winddirection == True:
+            u_df = self._reanalysis_aggregate[r + '_u'].to_frame().dropna() # Drop NA values from monthly/daily reanalysis data series
+            v_df = self._reanalysis_aggregate[r + '_v'].to_frame().dropna() # Drop NA values from monthly/daily reanalysis data series
+            u_data = u_df.tail(n * self._calendar_samples) # Get last 'x' years of data from reanalysis product
+            v_data = v_df.tail(n * self._calendar_samples) # Get last 'x' years of data from reanalysis product
+            u_aggregate = self.groupby_time_res(u_data)[r + '_u'] # Get long-term annualized monthly/daily wind components
+            v_aggregate = self.groupby_time_res(v_data)[r + '_v'] # Get long-term annualized monthly/daily wind components
+            wd_aggregate = 180-np.rad2deg(np.arctan2(-u_aggregate,v_aggregate)) # Calculate wind direction
+            wd_sin_aggregate = np.sin(2*np.pi*wd_aggregate/360) # Calculate sine of wind direction
+            wd_cos_aggregate = np.cos(2*np.pi*wd_aggregate/360) # Calculate cosine of wind direction
+            long_term_reg_inputs = pd.concat([long_term_reg_inputs.astype(float).reset_index(drop=True), wd_sin_aggregate.astype(float).reset_index(drop=True)], axis=1)
+            long_term_reg_inputs = pd.concat([long_term_reg_inputs.astype(float).reset_index(drop=True), wd_cos_aggregate.astype(float).reset_index(drop=True)], axis=1)
```

Comment:
Hi Jordan, thanks for suggesting these changes.
I have tried to implement them, but I have some difficulties related to the fact that while temperature is used "as-is" as regression input, for wind direction I actually incorporate its sine and cosine (to let the ML models better understand its cyclical nature, so that wd = 0 and wd = 359 do not appear as super different). 
Therefore, the calculations performed on temperature and wind direction are similar, though not identical.

For this piece of code, I was able to change it as follows. It's a bit longer than what you suggested, so please let me know if you have ideas on how to further shorten it:

`        # Temperature and wind direction
        namescol = [r + '_' + var for var in self._rean_vars]
        long_term_temp = self._reanalysis_aggregate[namescol].dropna().tail(n * self._calendar_samples).apply(self.groupby_time_res)
        if self.reg_temperature:
            long_term_reg_inputs = pd.concat([long_term_reg_inputs.astype(float).reset_index(drop=True), long_term_temp[r + '_temperature_K'].astype(float).reset_index(drop=True)], axis=1)        
        if self.reg_winddirection:
            wd_aggregate = np.pi-np.arctan2(-long_term_temp[r + '_u_ms'],long_term_temp[r + '_v_ms']) # Calculate wind direction
            long_term_reg_inputs = pd.concat([long_term_reg_inputs.astype(float).reset_index(drop=True), np.sin(wd_aggregate).astype(float).reset_index(drop=True)], axis=1)
            long_term_reg_inputs = pd.concat([long_term_reg_inputs.astype(float).reset_index(drop=True), np.cos(wd_aggregate).astype(float).reset_index(drop=True)], axis=1)
     `